### ISSUE TITLE
calendar_days_back set to 0 showing events in the past

### ISSUE THREAD

**Issue by kaizersoje** [gadgetchnnel/lovelace-home-feed-card#75]
I have set the `calendar_days_back `parameter to 0. However, the card displays events in past days.

Card version: 0.6.2
HA version: 2021.6.5

```
type: custom:home-feed-card
title: Calendar Feed
card_id: cal_feed
id_filter: ^home_feed_.*
calendars:
  - calendar.gmail_cal
  - calendar.home_calendar
  - calendar.facebook_events
calendar_days_forward: 0
calendar_days_back: 0
compact_mode: true
```

Screenshot below

![image](https://user-images.githubusercontent.com/54566217/123339103-551cfc00-d58d-11eb-905f-50c8c4823a13.png)

**Comment by gadgetchnnel** [comment]
What is the end time of the event?  I can't reproduce this, except for events which cross over from one day to the next (e.g. 10PM yesterday to 10AM today) which would be expected as the event was still active today.

**Comment by kaizersoje** [comment]
The event was from 2.30pm to 3.30 pm yesterday, so it did not cross into the current day .

**Comment by kaizersoje** [comment]
Closed it by mistake

**Comment by gadgetchnnel** [comment]
If you're not already on the latest version (0.6.2) can you please update to that (that version has logging of the version number to the browser console to make it easier to tell what version you're on) and give it another try?

If you still have the same issue, can you please let me know what time zone you're on?
The calendar event queries are done based on UTC and there may be an issue with how the time zone conversion is being applied.

**Comment by kaizersoje** [comment]
I am on version 0.6.2. I mentioned the version above. My timezone is AEST ( UTC +10).

**Comment by gadgetchnnel** [comment]
I found a potential issue with  how the times were converted to UTC to be passed to the APIs. The time was converted to UTC before applying calendar_days_back, which could cause issues depending on the timezone.

I have just release a new beta version, 0.6.3b1, where the time is only converted to UTC after calendar_days_back has been applied.

Can you please update to that version and let me know if it fixes it for you?

**Comment by kaizersoje** [comment]
Yes, it is fixed in v0.6.3b1. Thanks for the fix.
